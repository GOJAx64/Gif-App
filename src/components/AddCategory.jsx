import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  
  const [inputValue, setInputValue] = useState('');
  
  const onInputChange = ({ target }) => { // event.target.value
    setInputValue(target.value);
  }

  const onSubmit = (event) => {
    event.preventDefault();
    const input = inputValue.trim().toLocaleLowerCase();
    input.length < 1 ? null : onNewCategory(input);
    setInputValue('');
  }

  return (
    <form onSubmit={ onSubmit } aria-label="form">
      <input
        type="text"
        placeholder="Search Gifs"
        value={ inputValue }
        onChange= { onInputChange } // (event) => onInputChange(event)
      />
    </form>
  )
}

AddCategory.protoTypes = {
  onNewCategory: PropTypes.func.isRequired,
}