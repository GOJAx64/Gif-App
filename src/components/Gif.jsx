import PropTypes from 'prop-types';

export const Gif = ({ title, url }) => {
  return (
    <div className='card'>
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
  )
}

Gif.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
