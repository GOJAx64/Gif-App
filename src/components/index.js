export * from './AddCategory';
export * from './GifsGrid';
export * from './Gif';