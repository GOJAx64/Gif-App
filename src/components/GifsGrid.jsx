import { Gif } from './Gif';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifsGrid = ({ category }) => {
  
  const { gifs, isLoading } = useFetchGifs(category);
  
  return (
    <>
      <h3>{ category }</h3>
      { isLoading && <h2>Loading...</h2> }
      <div className='card-grid'>
        { 
          gifs.map( ( gif ) => 
            <Gif key={ gif.id } { ...gif } /> 
          )
        }
      </div>
    </>
  )
}

GifsGrid.propTypes = {
  category: PropTypes.string.isRequired,
}
