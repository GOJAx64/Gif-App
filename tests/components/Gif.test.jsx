import { render, screen } from '@testing-library/react';
import { Gif } from '../../src/components/Gif';

describe('Tests in <Gif />', () => { 

    const title = 'zelda';
    const url = 'https://zelda.com/link.jpg';

    test('should match with the snapshot', () => { 
        const { container } = render(<Gif title={ title } url={ url } />);
        expect(container).toMatchSnapshot();
    });

    test('should show the image with the URL and the indicated ALT', () => { 
        render(<Gif title={ title } url={ url } />);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url); // expect(screen.getByRole('img').src).toBe(url);
        expect(alt).toBe(alt); // expect(screen.getByRole('img').alt).toBe(title);
    });

    test('should show the title in the component', () => { 
        render(<Gif title={ title } url={ url } />);
        expect(screen.getByText(title)).toBeTruthy();
    });

});