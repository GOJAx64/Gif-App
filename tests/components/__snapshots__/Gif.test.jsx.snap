// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Gif /> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="zelda"
      src="https://zelda.com/link.jpg"
    />
    <p>
      zelda
    </p>
  </div>
</div>
`;
